src/main/webapp/script.js

```diff
@@ -0,0 +1,23 @@
+function initMap() {
+  var foodLocations = { 'mozerella': {lat: 1.326574, lng: 103.814158},
+                        'pastries': {lat: 1.322040, lng: 103.814554},
+                        'curry puffs': {lat: 1.323561, lng: 103.841931} };
+  const map = new google.maps.Map(document.getElementById("map"), {
+    center: { lat: 1.3521, lng: 103.8198 },
+    zoom: 14
+  });
+
+  var marker;
+  for (const [key, value] of Object.entries(foodLocations)) {
+    marker = new google.maps.Marker({position: value, map: map});
+    (function(marker, key) {
+      // add click event
+      google.maps.event.addListener(marker, 'click', function() {
+        infowindow = new google.maps.InfoWindow({
+            content: key
```

Comment:
Consider not using the key as the content but instead add another field to contain the content.
The key should be changed to describe the value.
And then you can iterate through the object one by one and access the keys.

Each marker should also point to a specific place, instead of a category, or else its use would be limited.

```
  var foodLocations = { 'position': {lat: 1.326574, lng: 103.814158}, 'place': 'Name of Business'} 
                        'position': {lat: 1.322040, lng: 103.814554}, 'place': 'Name of Business'}
                        'position': {lat: 1.323561, lng: 103.841931} }; ''place': 'Name of Business'};

for (i in foodLocations) {
    marker = new google.maps.Marker({position: i.position, map: map});
    (function(marker, i) {
      // add click event
      google.maps.event.addListener(marker, 'click', function() {
        infowindow = new google.maps.InfoWindow({
            content: i.desc
.
.
.
.

```